src/tools/generate/generate_figure_ids.js

```diff
@@ -1,8 +1,10 @@
 const generate_figure_ids = (html) => {
-  const re = /<figure>|<figure markdown>/gi;
+  const re = /<figure>|<figure data-markdown>/gi;
 
   let i = 1;
-  return html.replace(re, () => `<figure id='fig-${i++}'>`);
+  html = html.replace(re, () => `<figure id='fig-${i++}'>`);
+  const figcaption_regex = /<figcaption(.*?)>(.*?)([0-9]+)\./gi;
```

Comment:
Haven't tested but should the .* be changed to [^>]*?

Reply:
Which one? I've used non-greedy (`.*?`) and seems to work.